src/main/scala/cc/factorie/app/nlp/ner/StackedChainNer.scala

```diff
@@ -585,55 +585,10 @@ class StackedChainNer[L<:NerTag](labelDomain: CategoricalDomain[String],
   }
 }
 
-object StackedChainNer {
-  def initLexicons() : Unit = synchronized {
-    lexicon.iesl.Month.toString()
-    lexicon.iesl.Day.toString()
-
-    lexicon.iesl.PersonFirst.toString()
-    lexicon.iesl.PersonFirstHigh.toString()
-    lexicon.iesl.PersonFirstHighest.toString()
-    lexicon.iesl.PersonFirstMedium.toString()
-
-    lexicon.iesl.PersonLast.toString()
-    lexicon.iesl.PersonLastHigh.toString()
-    lexicon.iesl.PersonLastHighest.toString()
-    lexicon.iesl.PersonLastMedium.toString()
-
-    lexicon.iesl.PersonHonorific.toString()
-
-    lexicon.iesl.Company.toString()
-    lexicon.iesl.JobTitle.toString()
-    lexicon.iesl.OrgSuffix.toString()
-
-    lexicon.iesl.Country.toString()
-    lexicon.iesl.City.toString()
-    lexicon.iesl.PlaceSuffix.toString()
-    lexicon.iesl.USState.toString()
-    lexicon.iesl.Continents.toString()
-
-    lexicon.wikipedia.Person.toString()
-    lexicon.wikipedia.Event.toString()
-    lexicon.wikipedia.Location.toString()
-    lexicon.wikipedia.Organization.toString()
-    lexicon.wikipedia.ManMadeThing.toString()
-    lexicon.iesl.Demonym.toString()
-
-    lexicon.wikipedia.Book.toString()
-    lexicon.wikipedia.Business.toString()
-    lexicon.wikipedia.Film.toString()
-
-    lexicon.wikipedia.LocationAndRedirect.toString()
-    lexicon.wikipedia.PersonAndRedirect.toString()
-    lexicon.wikipedia.OrganizationAndRedirect.toString()
-  }
-}
-
```

Comment:
@eelstretching Most of the content of this PR goes to address the concerns that you outline in the last paragraph. Specifically, in the current state Factorie (as it is commonly used) is very opinionated about where resources (models and lexicons) get loaded from - they must be loaded from the classpath or pointed to as a URL in a System property. These are both very tightly coupled to the initialization conditions of the JVM which is highly suboptimal in a managed environment. the purpose of `cc.factorie.util.ModelProvider` is to allow users to specify something from java.io and have it used to provide trained models and lexicons. There are a few things that I want to add to make that even smoother, but as a start you can take a look at the scaladoc comments on `ModelProvider` to get a sense of the goal.

In terms of your specific concerns about loading and thread safety, in the case that you outlined above where you have two NER systems on separate threads the new functionality actually will provide separate lexicon instances to each of them (thus ensuring thread safety, but doubling memory usage). This is definitely suboptimal and I hadn't considered this case, so thanks for pointing it out. I think what I'll do to address it is to allow users to pass a `cc.factorie.app.nlp.lexicon.StaticLexicons` instance directly to NER class (or to lexicon using classes in general). In that case the user would initialize the `StaticLexicons` and 'load' (by just touching them to trigger scala lazy val loading) the lexicons they would use in a main thread then pass the `StaticLexicons` instance to each of their separate parallel NER systems which would share the resources. Any code that 'loaded' the lexicons in construction of the parallel instances would have no effect since they are already loaded.